docs/src/modules/spring/pages/value-entity.adoc

```diff
@@ -1 +1,112 @@
 = Implementing Value Entities
+
+include::ROOT:partial$include.adoc[]
+
+https://docs.kalix.io/reference/glossary.html#value_entity[Value Entities] persist state on every change and thus Kalix needs to serialize that data to send it to the underlying data store. However, we recommend that you do not persist your service's public API messages. This may introduce some overhead to convert from one type to an internal one but allows the service public interface logic to evolve independently of the data storage format, which should be private.
+
+The steps necessary to implement a Value Entity include:
+
+. Defining the API and model the entity's state.
+. Implementing behavior in command handlers.
+. Creating and initializing the Entity.
+
+The following sections walk through these steps using a counter service as an example.
+
+== Modeling the entity
+
+As mentioned above, to help us illustrate a Value Entity, we will be implementing a Counter service. For such service, we will want to be able to set the initial counter value but also to increase the counter modifying its state. The state will be a simple `Integer` but we will use a wrapper class `Number` for our API, as shown below:
+
+.src/main/java/com/example/Number.java
+[source,java]
+----
+include::example$spring-valueentity-counter/src/main/java/com/example/Number.java[]
+----
+
+
+== Implementing behavior
+
+Having a good idea of what we want to build and how we will interact with it, we will need to:
```

Comment:
This phrase is a bit confusing for me. I'm not sure if it's my lack of knowledge of English, though. The beginning is mentioning `Having a good idea` and  at the end `we will need to` doesn't seem connected in my mind.

Reply:
ahm, yeah.. not great indeed. let me revisit it

Reply:
fixed, see what you think now